### PR TITLE
Add Checkbox component

### PR DESCRIPTION
**Pull request by frozenhelium**
- Update spacing related CSS variables
- Add surface colors
- Add more option for spacing and font-size related variables

### CODE REVIEW

File: src/components/Checkbox/Checkmark/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+function Checkmark(props: CheckmarkProps) {
+    const {
+        className,
+        indeterminate,
+        value,
+    } = props;
+
+    const iconClassName = _cs(
+        className,
+    );
```
Comment (tnagorra) [inline src/components/Checkbox/Checkmark/index.tsx]: Why use _cs function here?

File: src/components/Checkbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    readOnly?: boolean;
+    indeterminate?: boolean;
+    uiMode?: UiMode;
+    tooltip?: string;
+    value: boolean | undefined | null;
+    onChange: (value: boolean, name: N) => void;
+    name: N;
+}
+
+function Checkbox<N extends string>(props: CheckboxProps<N>) {
```
Comment (tnagorra) [inline src/components/Checkbox/index.tsx]: Maybe we should also extend number here?

**Review by tnagorra — CHANGES_REQUESTED** [review]
Need to check generic type for Checkbox

**Comment by frozenhelium** [comment]
@tnagorra Done!!!

**Review by tnagorra — APPROVED** [review]

File: src/index.tsx [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import './styles.css';
 
+export type { Props as CheckboxProps } from './components/Checkbox';
+export { default as Checkbox } from './components/Checkbox';
```
Comment (AdityaKhatri) [inline src/index.tsx]: Plz sort

**Pull request merged**